js/print.js

```diff
@@ -0,0 +1,316 @@
+}
+
+/**
+ * Generates link elements for table of contents items
+ * */
+function renderTocLink(clauseID, clauseNumber, clauseTitle) {
+  const nonAnnexSections = [
+    'sec-copyright-and-software-license',
+    'sec-colophon',
+    'sec-bibliography',
+  ];
+  const link = document.createElement('a');
+  link.setAttribute('href', '#' + clauseID);
+  link.setAttribute('title', clauseTitle);
+
+  if (nonAnnexSections.includes(clauseID)) {
+    link.innerHTML = '<span class="secnum">' + clauseTitle + '</span>';
+
+    return link;
+  }
+
+  if (/^[A-Z]$/.test(clauseNumber)) {
+    const annexType = document.getElementById(clauseID).hasAttribute('normative')
+      ? 'normative'
+      : 'informative';
+    link.innerHTML =
+      '<span class="secnum">Annex ' +
+      clauseNumber +
+      ' <span class="unbold">(' +
+      annexType +
+      ')</span></span> ' +
+      clauseTitle;
+
+    return link;
+  }
+
+  if (/^[A-Z]\./.test(clauseNumber)) {
+    link.innerHTML = '<span class="secnum">Annex ' + clauseNumber + '</span> ' + clauseTitle;
+    return link;
+  }
+
+  link.innerHTML = '<span class="secnum">' + clauseNumber + '</span> ' + clauseTitle;
+  return link;
+}
+
+/**
+ * Loops through every clause/annex's h1 and improves the markup
+ * */
+function improveSectionHeadings() {
+  // eslint-disable-next-line prettier/prettier
+  const sectionHeadings = Array.from(specContainer.querySelectorAll('emu-clause > h1, emu-annex > h1'));
+
+  /** these section IDs are emu-annex elements but not functionally annexes */
+  const nonAnnexSections = [
+    'sec-copyright-and-software-license',
+    'sec-colophon',
+    'sec-bibliography',
+  ];
+
+  sectionHeadings.forEach(h1 => {
+    const secnum = h1.firstChild;
+    const section = secnum.innerHTML;
+    const parent = h1.parentNode;
+
+    if (/^[A-Z]$/.test(section)) {
+      h1.classList.add('annex-title');
+
+      if (nonAnnexSections.includes(parent.id)) {
+        secnum.innerHTML = '';
+      } else {
+        const annexType = parent.hasAttribute('normative') ? 'normative' : 'informative';
+
+        secnum.innerHTML =
+          'Annex ' + section + ' <br/><span class="unbold">(' + annexType + ')</span><br/>';
+      }
+    } else {
+      secnum.textContent = section;
+    }
+
+    if (secnum.textContent !== '') {
+      h1.insertBefore(document.createTextNode(' '), h1.firstChild);
+      h1.insertBefore(secnum, h1.firstChild);
+    }
+  });
+}
+
+/**
+ * The emu-imports element interferes with a ton of css specificity so we just
+ * take everything out and plop it directly into the DOM
+ * */
+function removeEmuImports() {
+  const emuImports = Array.from(specContainer.getElementsByTagName('emu-import'));
+
+  emuImports.forEach(importedMarkup => {
+    while (importedMarkup.hasChildNodes()) {
+      importedMarkup.parentNode.insertBefore(importedMarkup.childNodes[0], importedMarkup);
+    }
+    importedMarkup.parentNode.removeChild(importedMarkup);
+  });
+}
```

Comment:
I think it'd be fine to just not emit these in the first place.

Reply:
I’d prefer to either leave that in your capable hands, or tackle this myself in a separate PR. Most importantly, after all the time, effort, and ecma money I dumped into producing a quality document, I want to merge what worked.